### DISCUSSION TITLE
Qt Appimage crash reporting missing file when open file dialog (packed using linuxdeployqt)

### DISCUSSION THREAD

**Discussion by ychclone** [AppImage/AppImageKit#1323] [Q&A] [answered]
Hi, I've packaged my Qt application using linuxdeployqt:
[](https://sourceforge.net/projects/blink-code-search/files/1.9.1/)
[https://github.com/ychclone/blink/tree/master/blinkAppImage](url)

But it crashes when it open open dialog:

`(Blink-x86_64.AppImage:16591): Gtk-WARNING **: 10:37:40.688: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)

How do I include the missing file in the AppImage? Or could linuxdeployqt include it automatically?
I'm using Ubuntu 20 to run linuxdeployqt

`

**Comment by ychclone** [comment]
It crashes when it open file dialog

**Reply by probonopd** [reply]
Strangely this error seems Gtk related.
Are you deploying some Qt plugins that are usingGtk?

**Reply by ychclone** [reply]
No, i'm only using widget, network, xml in the pro file. The program doesn't crash when it starts. It only crash when it load a file open dialog. Please see below the qmake project file:

`TEMPLATE    = app
#CONFIG += release

CONFIG      += qscintilla2

#QMAKE_CXXFLAGS += -g -fno-omit-frame-pointer

QT += widgets xml network
RC_FILE = Resources/Icons/appIcons.rc
RESOURCES   = Resources/app.qrc
#RESOURCES   += Resources/webview.qrc 

INCLUDEPATH += "Utils"
INCLUDEPATH += "/opt/QScintilla_src-2.14.1/src"

#LIBS += -LC:/QScintilla_src-2.14.1/src/release -llibqscintilla2_qt6
#LIBS += -LC:/QScintilla_src-2.14.1/src/debug -llibqscintilla2_qt6d

unix {
LIBS += -L/opt/QScintilla_src-2.14.1/src -lqscintilla2_qt6
}
win32 {
LIBS += -LC:/QScintilla_src-2.14.1/src/release -llibqscintilla2_qt6
}

FORMS       = Resources/Forms/mainWindow.ui \
              Resources/Forms/editor.ui \
              Resources/Forms/projectDialog.ui \
              Resources/Forms/aboutDialog.ui \
              Resources/Forms/configDialog.ui \
              Resources/Forms/findReplaceDialog.ui \
              Resources/Forms/editorFindDialog.ui

SOURCES     += main.cpp \
               Utils/CUtils.cpp \
               Display/CMainWindow.cpp \
               Display/CEditor.cpp \
               Display/CEditorFindDlg.cpp \
               Display/CProjectDlg.cpp \
               Display/CAboutDlg.cpp \
               Display/CProjectListWidget.cpp \
               Display/CFileListWidget.cpp \
               Display/CConfigDlg.cpp \
               Display/CEventFilterObj.cpp \
               Display/CSearchTextBrowser.cpp \
               Display/CSearchTextEdit.cpp \
               Display/CFindReplaceDlg.cpp \
               Model/qTagger/CTagItem.cpp \
               Model/qTagger/CTagFileRecord.cpp \
               Model/qTagger/CTagResultItem.cpp \
               Model/qTagger/qTagger.cpp \
               Model/qTagger/CSourceFileList.cpp \
               Model/qFindReplacer/qFindReplacer.cpp \
               Model/CProjectListModel.cpp \
               Model/CFileListModel.cpp \
               Model/CProjectManager.cpp \
               Model/CProjectUpdateThread.cpp \
               Model/CProjectLoadThread.cpp \
               Model/CConfigManager.cpp \
               Model/CProjectItem.cpp \
               Model/CFileItem.cpp \
               Model/CRunCommand.cpp \ 
               Model/CFindReplaceModel.cpp \
               Storage/CXmlStorageHandler.cpp \


HEADERS     += Utils/commonType.h \
                Utils/CUtils.h \
                Display/CMainWindow.h \
                Display/CEditor.h \
                Display/CEditorFindDlg.h \
                Display/CProjectDlg.h \
                Display/CAboutDlg.h \
                Display/CProjectListWidget.h \
                Display/CFileListWidget.h \
                Display/CConfigDlg.h \
                Display/CEventFilterObj.h \
                Display/CSearchTextBrowser.h \
                Display/CSearchTextEdit.h \ 
                Display/CFindReplaceDlg.h \
                Model/qTagger/CTagItem.h \ 
                Model/qTagger/CTagFileRecord.h \
                Model/qTagger/CTagResultItem.h \
                Model/qTagger/qTagger.h \ 
                Model/qTagger/CSourceFileList.h \
                Model/qFindReplacer/qFindReplacer.h \
                Model/CProjectListModel.h \ 
                Model/CFileListModel.h \
                Model/CProjectManager.h \
                Model/CProjectUpdateThread.h \
                Model/CProjectLoadThread.h \
                Model/CConfigManager.h \
                Model/CProjectItem.h \ 
                Model/CFileItem.h \
                Model/CRunCommand.h \
                Model/CFindReplaceModel.h \
                Storage/CXmlStorageHandler.h \
                Storage/IStorageHandler.h \`

**Reply by probonopd** [reply]
Wonder what is giving the `Gtk:ERROR` then.

**Comment by probonopd** [comment]
Hi, I'd like to reproduce this but unfortunately there is no AppImage for download at https://github.com/ychclone/blink/releases.

**Reply by ychclone** [reply]
Hi, there's AppImage download in sourceforge:

https://sourceforge.net/projects/blink-code-search/files/1.9.1/

**Reply by probonopd** [reply]
Does not crash on my system when opening the file open dialog:

![image](https://github.com/AppImage/AppImageKit/assets/2480569/4e242ce2-6574-4aa0-ae8c-690956c9e557)

On my system there is no Gtk installed.

**Reply by probonopd** [reply, marked as answer]
The culprit is most likely that you are shipping Qt Gtk integration. Please try this:

```
./Blink-x86_64.AppImage --appimage-extract
./squashfs-root/AppRun # try it, it will crash at the file open dialog
rm squashfs-root/usr/lib/libgtk-3.so.0
rm squashfs-root/usr/plugins/platformthemes/libqgtk3.so
./squashfs-root/AppRun # try it, it will no longer crash at the file open dialog
```

So make sure your AppDir does NOT include

```
usr/lib/libgtk-3.so.0
usr/plugins/platformthemes/libqgtk3.so
```

You might get away with just removing `libgtk-3.so.0` and leaving `libqgtk3.so``, but ymmv.

**Reply by ychclone** [reply]
How can I exclude libgtk-3.so.0 in the build? I'm just using the directory below for the build:
https://github.com/ychclone/blink/tree/master/blinkAppImage

And the following .pro file for the build:
https://github.com/ychclone/blink/blob/master/blink.pro

And the following command:
~/linuxdeployqt-continuous-x86_64.AppImage  blinkAppImage/usr/share/applications/blink.desktop -verbose=2 -appimage -extra-plugins=iconengines,platformthemes/libqgtk3.so

**Reply by probonopd** [reply]
Leave away the `,platformthemes/libqgtk3.so`

**Reply by ychclone** [reply]
Hi, it still crash after removing ,platformthemes/libqgtk3.so

Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/Yaru/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted (core dumped)

**Reply by probonopd** [reply]
OK, so you most likely still have remnants of Gtk in your AppImage... can you check whether your AppImage does NOT include

```
usr/lib/libgtk-3.so.0
usr/plugins/platformthemes/libqgtk3.so
```

for sure?

**Reply by ychclone** [reply]
Yes these files are still there even without the use of the options ,platformthemes/libqgtk3.so

**Reply by probonopd** [reply]
So it is not fixed yet. You need to make sure that these files do not get into your AppDir/AppImage.
